eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/ImpTest.hs

```diff
@@ -938,10 +947,14 @@ trySubmitTx tx = do
   lEnv <- impLedgerEnv st
   ImpTestState {impRootTxIn} <- get
   res <- tryRunImpRule @"LEDGER" lEnv (st ^. nesEsL . esLStateL) txFixed
+  roundTripCheck <- asks iteFailuresRoundTrip
   case res of
     Left predFailures -> do
       -- Verify that produced predicate failures are ready for the node-to-client protocol
-      liftIO $ forM_ predFailures $ roundTripEraExpectation @era
+      when roundTripCheck $
+        liftIO $
+          forM_ predFailures $
+            roundTripEraExpectation @era
```

Comment:
It is better to ensure that there is a roundtrip failure, otherwise we might be either mistakenly disabling the check or we have already fixed the serialization, but forgot to adjust the test.
```suggestion
      liftIO $
        if roundTripCheck
          then
            forM_ predFailures $ roundTripEraExpectation @era
          else
            roundTripCborRangeFailureExpectation
              (eraProtVerLow @era)
              (eraProtVerHigh @era)
              predFailures
```
Note that we need to roundtrip the whole list, cause we can't specify which one of the failures is faulty

Reply:
Nice, thank you. In this way, I was forced to not skip the tests for Conway, where the serialization was fixed.